src/ostree/ot-builtin-commit.c

```diff
@@ -441,33 +441,32 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
   context = g_option_context_new ("[PATH]");
 
   if (!ostree_option_context_parse (context, options, &argc, &argv, invocation, &repo, cancellable, error))
-    goto out;
+    return glnx_prefix_error (error, "Parsing options");
```

Comment:
I think we should avoid mixiong `goto out` and `return` style in a single function because it's then easy to miss the manual cleanups when refactoring code.

In this case I think we can push this forward into fully using autocleanups with `_ostree_repo_auto_transaction_start (repo, cancellable, error);`?  

Oh I see, that's private API.  Blah.  Probably time to make it public.

Reply:
Oh sorry, I didn't consider that, only noticed a large chunks of not-strictly-needed gotos and tweaked them.

I've kept split commits, so can I drop that part from here. Or I can try to break+nest the logic here into multiple layers with separate cleanups (to keep the new error prefixes). I'd rather not delve into transaction changes at this point.

Reply:
Ah yep cool; I think I'd say let's be conservative and just do the second commit here, and split the first one (err prefixing) to a separate PR and investigate making the auto txn bits public.